src/java/htsjdk/variant/variantcontext/GenotypeJEXLContext.java

```diff
@@ -0,0 +1,58 @@
+package htsjdk.variant.variantcontext;
+
+import htsjdk.variant.vcf.VCFConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author bbimber
+ *
+ * implements the JEXL context for Genotype; this saves us from
+ * having to generate a JEXL context lookup map everytime we want to evaluate an expression.
+ *
+ */
+public class GenotypeJEXLContext extends VariantJEXLContext {
+    private Genotype g;
+
+    private interface AttributeGetter {
+        public Object get(Genotype g);
+    }
+
+    private static Map<String, AttributeGetter> attrs = new HashMap<String, AttributeGetter>();
+
+    static {
+        attrs.put("g", new AttributeGetter() { public Object get(Genotype g) { return g; }});
+        attrs.put(VCFConstants.GENOTYPE_KEY, new AttributeGetter() { public Object get(Genotype g) { return g.getGenotypeString(); }});
+
+        attrs.put("isHom", new AttributeGetter() { public Object get(Genotype g) { return g.isHom() ? "1" : "0"; }});
+        attrs.put("isHomRef", new AttributeGetter() { public Object get(Genotype g) { return g.isHomRef() ? "1" : "0"; }});
+        attrs.put("isHet", new AttributeGetter() { public Object get(Genotype g) { return g.isHet() ? "1" : "0"; }});
+        attrs.put("isHomVar", new AttributeGetter() { public Object get(Genotype g) { return g.isHomVar() ? "1" : "0"; }});
+        attrs.put("isCalled", new AttributeGetter() { public Object get(Genotype g) { return g.isCalled() ? "1" : "0"; }});
+        attrs.put("isNoCall", new AttributeGetter() { public Object get(Genotype g) { return g.isNoCall() ? "1" : "0"; }});
+        attrs.put("isMixed", new AttributeGetter() { public Object get(Genotype g) { return g.isMixed() ? "1" : "0"; }});
+        attrs.put("isAvailable", new AttributeGetter() { public Object get(Genotype g) { return g.isAvailable() ? "1" : "0"; }});
+        attrs.put("isPassFT", new AttributeGetter() { public Object get(Genotype g) { return g.isFiltered() ? "0" : "1"; }});
+        attrs.put(VCFConstants.GENOTYPE_FILTER_KEY, new AttributeGetter() { public Object get(Genotype g) { return g.isFiltered()?  g.getFilters() : "PASS"; }});
+        attrs.put(VCFConstants.GENOTYPE_QUALITY_KEY, new AttributeGetter() { public Object get(Genotype g) { return g.getGQ(); }});
```

Comment:
these would be captured by GenotypeJEXLContext.get(), correct?  DP could be added to the static map, though I'm not sure how one supports extended (variable) attributes there.  

I was assuming the primary point of the map was to support non-standard keys (i.e. isHom, isHomRef).  Otherwise, we defer back to Genotype.hasAnyAttribute().  This is comparable to what exists in VariantJEXLContext.get().